sdk/core/Azure.Core.Experimental/tests/Variant/MemoryWatch.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using NUnit.Framework;
+
+namespace Azure
+{
+    public ref struct MemoryWatch
+    {
+        private static bool s_jit;
+        private long _allocations;
+
+        public static void JIT()
+        {
+            if (s_jit)
+            {
+                return;
+            }
+
+            // JITing allocates, so make sure we've got all of our <T> methods created.
+
+            Value.Create((bool)default).As<bool>();
```

Comment:
Should this also create an ArraySegment?

Reply:
@jeremykuhne, I noticed the array tests don't validate no allocations. Is this by design?